### PR TITLE
update cassandraIdx schema to parition based on Partition Id

### LINKED ISSUE #453: poor performance of querying metric_idx
The new metric_idx table is performing terribly.

We need to drop the clustering key as it is redundant.

```
The Partition Key is responsible for data distribution across your nodes.
The Clustering Key is responsible for data sorting within the partition.
The Primary Key is equivalent to the Partition Key in a single-field-key table.
The Composite/Compound Key is just a multiple-columns key
```

As our paritionKey is unique (metricDef.Id), there is 0 value to have a clusteringKey.

### PR DESCRIPTION
**Pull request by woodsaj**
fixes #453

### CODE REVIEW

File: docs/cassandra.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ CREATE TABLE IF NOT EXISTS raintank.metric_idx (
     mtype text,
     tags set<text>,
     lastupdate int,
-    PRIMARY KEY (id, partition)
+    PRIMARY KEY (partition, partition)
```
Comment (Dieterbe) [inline docs/cassandra.md]: typo?

**Comment by Dieterbe** [comment]
should we not update the query in `(c *CasIdx) Delete()` also ?

File: docs/cassandra.md (hunk 1 above)
Comment (woodsaj) [inline docs/cassandra.md]: yes.  And fixed.

**Comment by woodsaj** [comment]
> should we not update the query in (c *CasIdx) Delete() also ?

it has been updated.

File: docs/cassandra.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ CREATE TABLE IF NOT EXISTS raintank.metric_idx (
     mtype text,
     tags set<text>,
     lastupdate int,
-    PRIMARY KEY (id, partition)
+    PRIMARY KEY (partition, id)
 ) WITH compaction = {'class': 'SizeTieredCompactionStrategy'}
     AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'};
 CREATE INDEX IF NOT EXISTS ON raintank.metric_idx(partition);
```
Comment (Dieterbe) [inline docs/cassandra.md]: we should also remove all these `CREATE INDEX` on partition field right?

**Comment by Dieterbe** [comment]
I think I just updated `(c *CasIdx) Delete()` correctly in the above commit (you probably just mistook Prune for Delete).  However I don't quite understand the scenario of a metric being moved into a different partition. the input calls `in.metricIndex.Add(metric, partition)` which for both Cass/Es first call memoryIdx.AddDef (which will correctly replace the entry in the memory index) but then they just execute the write of the new def to cass/es. Will this not leave the def entry with the outdated (but identical otherwise and hence redundant) field in place? It looks like we should execute an explicit delete for the given id and the old partition.

**Comment by Dieterbe** [comment]
* the last commit logs the same error multiple times. within deleteDef, and in the caller of it.
* we should either also fix this for ES now, or create a ticket to do it later (up to you)

**Comment by woodsaj** [comment]
@Dieterbe error logs include the file and line the error is emittedfrom.
ES doesnt need changes, this is just fixing **Cassandra** schema

**Comment by Dieterbe** [comment]
FTR ES doesn't need the same change because it indexes by id and will do the overwrite.

**Pull request merged**